### PR TITLE
[KEYCLOAK-9666] - Entitlement request with service account results in server error

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

**Review by mhajas — DISMISSED** [review]
LGTM, thanks @pedroigor!

**Review by sguilhen — DISMISSED** [review]
LGTM except for the weird else statement I've highlighted.

File: authz/client/src/main/java/org/keycloak/authorization/client/resource/AuthorizationResource.java [hunk 1]
```diff
@@ -74,7 +74,11 @@ public AuthorizationResponse authorize(final AuthorizationRequest request) throw
         Callable<AuthorizationResponse> callable = new Callable<AuthorizationResponse>() {
             @Override
             public AuthorizationResponse call() throws Exception {
-                request.setAudience(configuration.getResource());
+                if (request.getAudience() == null) {
+                    request.setAudience(configuration.getResource());
+                } else {
+                    request.setAudience(request.getAudience());
```
[the remark this answers is not in the document]
Comment (pedroigor) [reply authz/client/src/main/java/org/keycloak/authorization/client/resource/AuthorizationResource.java]: It is not. Removed ...

**Review by mhajas — DISMISSED** [review]

**Review by sguilhen — DISMISSED** [review]

**Review by stianst — DISMISSED** [review]

**Comment by stianst** [comment]
@pedroigor needs a rebase

**Comment by pedroigor** [comment]
@stianst rebased

**Review by mhajas — APPROVED** [review]

**Review by sguilhen — APPROVED** [review]

**Review by stianst — APPROVED** [review]

**Pull request merged**